File: modules/@angular/forms/test/spies.dart
library core.spies;

import 'package:angular2/common.dart';
import 'package:angular2/src/core/change_detection/change_detection.dart';
import 'package:angular2/testing_internal.dart';

@proxy
class SpyNgControl extends SpyObject implements NgControl {}

@proxy
class SpyValueAccessor extends SpyObject implements ControlValueAccessor {}

@proxy
class SpyChangeDetectorRef extends SpyObject implements ChangeDetectorRef {}

File: modules/@angular/common/test/spies.ts
import {ChangeDetectorRef} from '@angular/core/src/change_detection/change_detector_ref';
import {SpyObject, proxy} from '@angular/core/testing/testing_internal';

export class SpyChangeDetectorRef extends SpyObject {
  constructor() {
    super(ChangeDetectorRef);
    this.spy('markForCheck');
  }
}

export class SpyNgControl extends SpyObject {}

export class SpyValueAccessor extends SpyObject { writeValue: any; }
